pkg/providers/smi/provider.go

```diff
@@ -77,6 +88,8 @@ func New(defaultMode string, tcpStateTable *k8s.State, ignored k8s.IgnoreWrapper
 
 // BuildConfig builds the configuration for routing
 // from a native kubernetes environment.
+// FIXME: Reduce cognitive complexity.
```

Comment:
Is this FIXME required at this point?

This is our one method that is big :(

Reply:
It's not required, but we felt really bad about adding the `// nolint:gocognit,gocyclo`. So we wanted to add this FIXME, to make sure we never ignore this `nolint` and make it clear this function should be refactored. WDYT?

Reply:
10-4